src/osdc/Objecter.cc

```diff
@@ -2310,22 +2310,32 @@ void Objecter::_op_submit(Op *op, shunique_lock& sul, ceph_tid_t *ptid)
 
   assert(op->target.flags & (CEPH_OSD_FLAG_READ|CEPH_OSD_FLAG_WRITE));
 
+  /*
+   * For FULL_FORCE/FULL_TRY, it can set directly on OP or set by Objcter
+   * like osdmap_full_force & osdmap_full_try
+   */
+  int flags = op->target.flags;
+  if (osdmap_full_force)
+    flags |= CEPH_OSD_FLAG_FULL_FORCE;
+  if (osdmap_full_try)
+    flags |= CEPH_OSD_FLAG_FULL_TRY;
+
```

Comment:
i think this should just set the bit in op->target.flags, and _prepare_osd_op() can drop the part the sets it when sending.